rye/tests/test_toolchain.rs

```diff
@@ -0,0 +1,37 @@
+use crate::common::{rye_cmd_snapshot, Space};
+
+mod common;
+
+#[test]
+fn test_fetch() {
```

Comment:
Pretty sure this test should be super flaky because it manipulates a python version used elsewhere. Can you make it use a version of 3.8 that is not used by other tests and leave a comment?

Reply:
Python 3.8 is nearing the end of its life cycle in just 7 months and could soon be unsupported by Rye. So I changed it to 3.12.1, which is not used in the test bootstrap, and remains support for long time enough.